Release 7.1.0 - Add tags in asg-ebs and asg-efs; fix vpc

dalenewby opened the pull request:
### Added
- Add optional user-supplied tags functionality to asg-ebs and asg-efs that already existed in asg.
- Add optional ECR lifecycle policy in ecr module.

### Fixed
- Include `vpc = true` in VPC resource `aws_eip` to fix a problem due to the retirement of EC2-Classic. The error message was "Error: with the retirement of EC2-Classic no new non-VPC EC2 EIPs can be created".

For release as 7.1.0

briskt committed:
set vpc = true in creation of elastic IP for NAT gateway

This is because of the retirement of IC2-Classic. Error message was "Error: with the retirement of EC2-Classic no new non-VPC EC2 EIPs can be created"

dalenewby committed:
Add optional tags for ASG

Add optional user-supplied tags functionality to asg-ebs and asg-efs that
already existed in asg.

dalenewby committed:
Merge pull request #59 from silinternational/tags

Add optional tags for ASG

briskt commented on aws/asg-ebs/README.md:
```diff
@@ -36,6 +36,7 @@ an auto scaling group that uses the template.  An EBS file system is mounted.
  - `ebs_mkfs_extraopts` - Extra options to pass to the mkfs command. Default: ""
  - `ebs_fs_type` - Type of filesystem to create. Default: `ext4`
  - `ebs_mountopts` - Mount options to include in /etc/fstab, default is "defaults,noatime"
+ - `tags` - A list of tag definitions in JSON format to be applied to the asg.
```
```suggestion
 - `tags` - A list of tag definitions to be applied to the asg. See example below.
```

briskt commented on aws/asg-efs/README.md:
```diff
@@ -27,6 +27,7 @@ an auto scaling group that uses the template.  An EFS file system is mounted.
  - `additional_security_groups` - List of additional security groups (in addition to default vpc security group)
  - `associate_public_ip_address` - true/false - Whether or not to associate public ip addresses with instances. Default: false
  - `additional_user_data` - command to append to the EC2 user\_data, default is ""
+ - `tags` - A list of tag definitions in JSON format to be applied to the asg.
```
```suggestion
 - `tags` - A list of tag definitions to be applied to the asg. See example below.
```

briskt commented on aws/asg-ebs/vars.tf:
```diff
@@ -77,6 +77,12 @@ variable "additional_user_data" {
 }
 
+variable "tags" {
+  type        = list(any)
```
```suggestion
  type        = list(map(any))
```

briskt commented on aws/asg-efs/vars.tf:
```diff
@@ -78,3 +78,8 @@ variable "additional_user_data" {
 }
 
+variable "tags" {
+  type        = list(any)
```
```suggestion
  type        = list(map(any))
```

briskt reviewed (dismissed):
Looks good. Just noted a few optional changes. If you want to go back and update the asg module in the same way, that would be fine.

Schparky commented:
One more thing...rather than continue using `tags` as a tuple (list), why not use a map like the aws provider uses? Then the next major version bump we can revise the existing `tags` that are lists to use a map as well. Or just do the major version bump now.

briskt committed:
remove timestamp recovery_point_tag from backup module

because it causes the state to be updated on every run, and this seems like not a useful tag to have

briskt committed:
Revert "remove timestamp recovery_point_tag from backup module"

This reverts commit 18cb6b27730194753b14d43c62cd8d8d951c7d14.

briskt committed:
Merge branch 'develop' into eip

briskt committed:
Merge pull request #62 from silinternational/eip

set vpc = true in creation of elastic IP for NAT gateway

briskt committed:
Merge pull request #61 from silinternational/feature/remove-timestamp

remove timestamp recovery_point_tag from backup module

briskt committed:
add tags to repository in aws/ecr module

briskt committed:
add default tag values

briskt committed:
add lifecycle policy to repository in aws/ecr module

briskt committed:
include the ecr policy in the main.tf file

briskt committed:
include the retention policy in the ecr module

briskt committed:
use the policy not the count

briskt committed:
need to use "any" since no tag prefix list is given

briskt committed:
add variable image_retention_tags and add more documentation in README

briskt committed:
Further clarification in README and comments

briskt committed:
Merge pull request #64 from silinternational/feature/ecr-lifecycle

add ECR lifecycle policy

briskt committed:
remove aws tags config

briskt committed:
Merge branch 'feature/ecr-tags' into develop

briskt committed:
correct the type of the `tags` variable in asg modules
